src/stylesheets/components/_icon.scss

```diff
@@ -16,6 +16,18 @@ $icon-inline-size-magic-number: 1em;
   height: $icon-inline-size-magic-number;
   position: relative;
   width: $icon-inline-size-magic-number;
+
+  .usa-button > &:first-child {
+    float: left;
+    margin-top: -0.5 * ($icon-inline-size-magic-number - lh($theme-button-font-family, 1));
+    margin-bottom: -0.5 * ($icon-inline-size-magic-number - lh($theme-button-font-family, 1));
+    margin-right: 0.25rem;
+  }
+
+  .usa-button:not(.usa-button--unstyled) > &:first-child {
+    margin-left: -0.5rem;
+    margin-right: 0.5rem;
```

Comment:
> I'm open to feedback about the specific offsets here, since it's all a bit subjective.

I adjusted this further in 5abd934 after some discussion with our in-team UX folks, based on differences in appearance across icons. Previously, the idea was to try to create a "centered" effect between the left edge of the button and the start of the text. With 5abd934, the idea is more to have the icon adjacent to the text, with a bit of margin. This is offset (negated) a bit based on some built-in padding to most icons, which appears to be _at least_ 2px inside the default 24px viewport. The idea with the calculation in 5abd934 is to adjust this relative to the font size of the button, so that the spacing can increase a bit with larger buttons (e.g. the "Big" variant).

`develop`|[`dd973bd`](https://github.com/uswds/uswds/pull/4493/commits/dd973bd33f3eab11be92349d286b04e83906fe83) (original proposal)|[`5abd934`](https://github.com/uswds/uswds/pull/4493/commits/5abd934869924b055a2292048be369b149729a15) (revised proposal)
---|---|---
![Screen Shot 2022-02-07 at 1 29 11 PM](https://user-images.githubusercontent.com/1779930/152854787-31a0fbd5-ee6c-41df-bf60-3c9363d1b661.png)|![Screen Shot 2022-02-07 at 1 28 13 PM](https://user-images.githubusercontent.com/1779930/152854786-1915e265-e0e4-46ba-8b5f-205c549ba72d.png)|![Screen Shot 2022-02-07 at 1 23 47 PM](https://user-images.githubusercontent.com/1779930/152854785-3cc3ae42-3380-4db2-8b7c-7ba69a9b5770.png)